sdk/core/Azure.Core/src/Pipeline/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,341 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Copied from https://raw.githubusercontent.com/dotnet/corefx/master/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpEnvironmentProxy.cs
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpEnvironmentProxyCredentials : ICredentials
```

Comment:
Since this is occupying a `System` namespace, should we consider placing it somewhere that makes that apparent?

Reply:
Why is it in System namespace?

Reply:
I'll move it to our namespace.

Reply:
@KrzysztofCwalina this file is a direct copy from corefx with very minor changes. I didn't modify it or clean it up to allow changes to be easily applied if required.